lib/Models/WebMapTileServiceCatalogItem.ts

```diff
@@ -0,0 +1,710 @@
+        ) {
+          baseUrl = resourceUrl.template;
+        }
+      }
+    }
+
+    const tileMatrixSet = this.tileMatrixSet;
+    if (!isDefined(tileMatrixSet)) {
+      return;
+    }
+
+    let rectangle;
+
+    if (
+      this.rectangle !== undefined &&
+      this.rectangle.east !== undefined &&
+      this.rectangle.west !== undefined &&
+      this.rectangle.north !== undefined &&
+      this.rectangle.south !== undefined
+    ) {
+      rectangle = Rectangle.fromDegrees(
+        this.rectangle.west,
+        this.rectangle.south,
+        this.rectangle.east,
+        this.rectangle.north
+      );
+    } else {
+      rectangle = undefined;
+    }
+
+    const imageryProvider = new WebMapTileServiceImageryProvider({
+      url: proxyCatalogItemUrl(this, baseUrl),
+      layer: layerIdentifier,
+      style: this.style,
+      tileMatrixSetID: tileMatrixSet.id,
+      tileMatrixLabels: tileMatrixSet.labels,
+      minimumLevel: tileMatrixSet.minLevel,
+      maximumLevel: tileMatrixSet.maxLevel,
+      tileWidth: tileMatrixSet.tileWidth,
+      tileHeight: tileMatrixSet.tileHeight,
+      tilingScheme: new WebMercatorTilingScheme(),
+      format,
+      rectangle
+    });
+    return imageryProvider;
+  }
+
+  @computed
+  get tileMatrixSet():
+    | {
+        id: string;
+        labels: string[];
+        maxLevel: number;
+        minLevel: number;
+        tileWidth: number;
+        tileHeight: number;
+      }
+    | undefined {
+    const stratum = <WmtsCapabilitiesStratum>(
+      this.strata.get(GetCapabilitiesMixin.getCapabilitiesStratumName)
+    );
+    if (!this.layer) {
+      return;
+    }
+    const layer = stratum.capabilitiesLayer;
+    if (!layer) {
+      return;
+    }
+
+    const usableTileMatrixSets = stratum.usableTileMatrixSets;
+
+    let tileMatrixSetLinks: TileMatrixSetLink[] = [];
+    if (layer?.TileMatrixSetLink) {
+      if (Array.isArray(layer?.TileMatrixSetLink)) {
+        tileMatrixSetLinks = [...layer?.TileMatrixSetLink];
+      } else {
+        tileMatrixSetLinks = [layer.TileMatrixSetLink];
+      }
+    }
+
+    let tileMatrixSetId: string =
+      "urn:ogc:def:wkss:OGC:1.0:GoogleMapsCompatible";
+    let maxLevel: number = 0;
+    let minLevel: number = 0;
+    let tileWidth: number = 256;
+    let tileHeight: number = 256;
+    let tileMatrixSetLabels: string[] = [];
+    let found = false;
+    for (let i = 0; i < tileMatrixSetLinks.length; i++) {
+      const tileMatrixSet = tileMatrixSetLinks[i].TileMatrixSet;
+      if (usableTileMatrixSets && usableTileMatrixSets[tileMatrixSet]) {
+        found = true;
+        tileMatrixSetId = tileMatrixSet;
+        tileMatrixSetLabels = usableTileMatrixSets[tileMatrixSet].identifiers;
+        tileWidth = Number(usableTileMatrixSets[tileMatrixSet].tileWidth);
+        tileHeight = Number(usableTileMatrixSets[tileMatrixSet].tileHeight);
+        break;
+      }
+    }
+
+    if (!found) return;
```

Comment:
I have added this check - and also a warning which will display if no tileMatrixSet is found.
Does this affect any of your layers?

Reply:
Warning looks like this - 
![image](https://user-images.githubusercontent.com/6187649/94011921-7036c180-fdeb-11ea-9638-677e51fa3b3c.png)

Reply:
Ok so my changes are breaking a few layers - for example  https://gibs.earthdata.nasa.gov/wmts/epsg3857/best/wmts.cgi

Reply:
I have reverted this